### ISSUE TITLE
ShowToast - Prevent "YOUR FLOW FINISHED" screen

### ISSUE THREAD

**Issue by Garywoo** [alexed1/LightningFlowComponents#958]
I have built a screenflow to be launched using an action button, which, under normal circumstances should simply complete an action, and then show the user a success toast message using the ShowToast action, without actually using any screen elements. If the flow encounters an error, this will be presented to the user with a screen element.

Currently, if no screen has been shown, and the flow finishes (with or without this ShowToast action) a screen with the message "YOUR FLOW FINISHED" will be shown to the user. There are no Next / Previous buttons on this screen, and the X button in the upper right must be used to close it. This is default Salesforce behaviour. The toast message is shown above this screen.

![image](https://user-images.githubusercontent.com/10392194/140101810-6ffafa4e-423d-4da3-99ec-ee7973b73da1.png)

It would be great if this action (or another action) could prevent or suppress this "YOUR FLOW FINISHED" screen, so that no further interactions are required by my users running the flow, and the modal screenflow window closes as normal when using the ShowToast action as the last flow action.

Preventing this screen is a popular topic on the Salesforce Ideas board, as can be seen here: https://ideas.salesforce.com/s/idea/a0B8W00000GdeS5UAJ/prevent-your-flow-finished-in-screen-flows

**Comment by ericrsmith35** [comment]
You could always move the button to a flow screen, thus avoiding the flow finished message.

**Comment by Garywoo** [comment]
@ericrsmith35 Having a success screen with a finish button is currently the workaround I'm using. 

Ideally, I don't want the user to see any screens at all if the flow completes successfully, only a toast. This would save the user a click and seeing a pointless screen. Screens are only required for error scenarios in this screenflow.

**Comment by jdpond** [comment]
+1 @tomSny and I have talked about doing this for another client too.

**Comment by ericrsmith35** [comment]
There wouldn't be an extra click nor a pointless screen if the button was
presented in the flow rather than on an action button.

**Comment by ericrsmith35** [comment]
Yes, it will change where the button is on the page.  Add a screen as the
first node in your flow, disable the footer, use the NavigationButton
component from unofficialsf.com to display your button on the screen.
Configure the button to just navigate to the next node in the flow.  Add
the flow to the page wherever you want the button to be.  When the flow
finishes and the toast message has displayed, the flow will automatically
restart and continue to show the button.

**Comment by Garywoo** [comment]
In that case, the user would still have to interact with a screen. I'm trying to avoid that being a requirement here, and having a one-click process to run the flow and then have it close.

**Comment by ericrsmith35** [comment]
There would be no changes in clicks by the user.  The flow IS the button.

**Comment by Garywoo** [comment]
Oh, are you suggesting embedding the flow on the lightning page as a component, and then running the action from that flow?  
  
Hmm, that could work, however we have a standard operating procedure of launching these flows from record action buttons at the top of the page, and already have a number of other actions in that area. Switching things up for this one other action could be confusing for users, as they'd have to remember to look in two places for taking actions on the record.

I'd prefer to keep things standardised here, and just launch the flow using the regular action buttons. Is there any possibility of preventing or suppressing this finished screen in that situation?

**Comment by ericrsmith35** [comment]
Not yet, but I voted for the idea.
